cmd/internal/cli/build_linux.go

```diff
@@ -39,6 +40,10 @@ import (
 
 func fakerootExec(isDeffile bool) {
 	useSuid := starter.IsSuidInstall()
+	if buildArgs.ignoreSubuid {
+		useSuid = false
+		os.Setenv("APPTAINER_IGNORE_SUBUID", "1")
```

Comment:
I'm sorry, I didn't mean to say that you needed to set these environment variables in the main apptainer code.  I was thinking that you would set them in the e2e code.  From e2e you instead pass the options, which is a fine way to do it.  Then there's no need to set the environment variables at all, because nothing is looking at them; the environment variables are just another way to set the options.  (In fact I believe the option handler will set the environment variable if the option is passed, in addition to reading the environment variable, but that doesn't matter).

Also, the option relevant here is userns, not ignoreSubuid.

Here you could simply do 
```
    useSuid := starter.IsSuidInstall() && !buildArgs.userns
```